core/domain/platform_parameter_domain_test.py

```diff
@@ -81,3 +82,1099 @@ def test_to_dict(self):
+            },
+        )
+        context.validate()
+
+    def test_validate_with_invalid_client_type_raises_exception(self):
+        context = parameter_domain.EvaluationContext.from_dict(
+            {
+                'client_type': 'invalid',
+                'browser_type': None,
+                'app_version': '1.0.0',
+                'user_locale': 'en',
+            },
+            {
+                'server_mode': 'dev',
+            },
+        )
+        with self.assertRaisesRegexp(
+            utils.ValidationError, 'Invalid client type \'invalid\''):
+            context.validate()
+
+    def test_validate_with_invalid_browser_type_raises_exception(self):
+        context = parameter_domain.EvaluationContext.from_dict(
+            {
+                'client_type': 'Web',
+                'browser_type': 'Invalid',
+                'app_version': '1.0.0',
+                'user_locale': 'en',
+            },
+            {
+                'server_mode': 'dev',
+            },
+        )
+        with self.assertRaisesRegexp(
+            utils.ValidationError, 'Invalid browser type \'Invalid\''):
+            context.validate()
+
+    def test_validate_with_invalid_app_version_raises_exception(self):
+        context = parameter_domain.EvaluationContext.from_dict(
+            {
+                'client_type': 'Android',
+                'browser_type': None,
+                'app_version': 'a.a.a',
+                'user_locale': 'en',
+            },
+            {
+                'server_mode': 'dev',
+            },
+        )
+        with self.assertRaisesRegexp(
+            utils.ValidationError, 'Invalid version \'a.a.a\''):
+            context.validate()
+
+    def test_validate_with_invalid_user_locale_raises_exception(self):
+        context = parameter_domain.EvaluationContext.from_dict(
+            {
+                'client_type': 'Android',
+                'browser_type': None,
+                'app_version': '1.0.0',
+                'user_locale': 'invalid',
+            },
+            {
+                'server_mode': 'dev',
+            },
+        )
+        with self.assertRaisesRegexp(
+            utils.ValidationError, 'Invalid user locale \'invalid\''):
+            context.validate()
+
+    def test_validate_with_invalid_server_mode_raises_exception(self):
+        context = parameter_domain.EvaluationContext.from_dict(
+            {
+                'client_type': 'Android',
+                'browser_type': None,
+                'app_version': '1.0.0',
+                'user_locale': 'en',
+            },
+            {
+                'server_mode': 'invalid',
+            },
+        )
+        with self.assertRaisesRegexp(
+            utils.ValidationError, 'Invalid server mode \'invalid\''):
+            context.validate()
+
+
+class PlatformParameterFilterTests(test_utils.GenericTestBase):
+    """Test for the PlatformParameterFilter."""
+
+    def create_example_context(
```

Comment:
Helper methods are generally named with a ‘_’ prefix to indicate that they are meant for private access.

Reply:
Done, thanks!